foundry/applications/formApplications/setupConfigurationForm.d.ts

```diff
@@ -1,3 +1,206 @@
+   */
+  protected _onInstallPackageDialog(event: JQuery.ClickEvent): Promise<void>;
+
+  /**
+   * Handle update button press for a single Package
+   */
+  protected _onUpdatePackage(
+    event: JQuery.ClickEvent
+  ):
+    | Promise<void>
+    | ReturnType<SetupConfigurationForm['_updateCheckOne']>
+    | ReturnType<SetupConfigurationForm['_updateDownloadOne']>;
+
+  /**
+   * Execute upon an update check for a single Package
+   * @param type   - The package type to check
+   * @param name   - The package name to check
+   * @param button - The update button for the package
+   */
+  protected _updateCheckOne(type: string, name: string, button: HTMLElement): Promise<void | boolean>;
+
+  /**
+   * Execute upon an update download for a single Package
+   * Returns a Promise which resolves once the download has successfully started
+   * @param type   - The package type to install
+   * @param name   - The package name to install
+   * @param button - The Download button
+   */
+  protected _updateDownloadOne(
+    type: string,
+    name: string,
+    button: HTMLElement
+  ): Promise<ReturnType<typeof SetupConfiguration['installPackage']>>;
+
+  /**
+   * Handle uninstall button clicks to remove existing packages
+   */
+  protected _onUninstallPackage(event: JQuery.ClickEvent): void;
+
+  /**
+   * Execute upon an update-all workflow to update all packages of a certain type
+   */
+  protected _onUpdatePackages(event: JQuery.ClickEvent): Promise<void>;
+
+  /**
+   * Handle button clicks to update the core VTT software
+   */
+  protected _onCoreUpdate(event: JQuery.ClickEvent): Promise<void>;
+
+  /**
+   * Activate socket listeners related to the Setup Configuration form
+   */
+  activateSocketListeners(): void;
+
+  /**
+   * Update the display of an installation progress bar for a particular progress packet
+   * @param data - The progress update data
+   */
+  protected _updateProgressBar(data: SetupConfigurationForm.ProgressData): void;
+
+  /**
+   * Update installation progress for a particular button which triggered the action
+   * @param data - The progress update data
+   */
+  protected _updateProgressButton(data: SetupConfigurationForm.ProgressData): void;
+
+  /**
+   * @remarks This method does not exist on SetupConfigurationForm and only exists to make the typescript compile!
+   */
+  protected _updateObject(...args: unknown[]): never;
+}
+
+declare namespace SetupConfigurationForm {
+  interface Data {
+    coreVersion: Game['data']['version'];
+    coreVersionHint: string;
+    systems: Data.AvailTagged<Game.System>[];
+    modules: Data.AvailTagged<Game.Module>[];
+    worlds: Data.AvailTagged<Game.World>[];
+    languages: Game['data']['languages'];
+    options: Game['data']['options'] & { upnp: boolean };
+    adminKey: Game['data']['adminKey'];
+    world: ReturnType<SetupConfigurationForm['worlds']['find']>;
+    current: undefined; // TODO: this seems to have a value that is never assigned (`this.current`)
```

Comment:
Yeah, seems like that's the case. Maybe create a bug report at foundry?

Reply:
This is now tracked by this upstream issue: https://gitlab.com/foundrynet/foundryvtt/-/issues/4772